src/main/java/org/folio/rest/impl/InvoiceHelper.java

```diff
@@ -319,7 +319,8 @@ private void validateInvoiceLineFundDistributions(List<InvoiceLine> invoiceLines
       .thenApply(voucher -> {
         invoice.setVoucherNumber(voucher.getVoucherNumber());
         return withRequiredFields(voucher, invoice);
-      });
+      })
+      .thenApply(createdVoucher -> createdVoucher.withAcqUnitIds(invoice.getAcqUnitIds()));
```

Comment:
It is not necessary to add new thenApply block, it can be done in the previous one.